src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs

```diff
@@ -1,11 +1,47 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class ICompilationExtensions
     {
+        public static ImmutableArray<Compilation> GetReferencedCompilations(this Compilation compilation)
+        {
+            var builder = ArrayBuilder<Compilation>.GetInstance();
```

Comment:
When I wrote it, I was assuming it would become a compiler helper.  There's only one caller, so that won't be necessary.  It will, however be enumerated immediately (and completely, I believe).